src/domain/alerts/alerts.repository.ts

```diff
@@ -51,33 +52,69 @@ export class AlertsRepository implements IAlertsRepository {
       return this._notifyUnknownTransaction(chainId, log);
     }
 
-    const safe = await this.safeRepository.getSafe({
-      chainId,
-      address: decodedEvent.args.to,
+    try {
+      const safeAddress = decodedEvent.args.to;
+
+      const newSafe = await this.predictNewSafeSetup({
+        chainId,
+        safeAddress,
+        decodedTransactions,
+      });
+
+      return this._notifyNewSafeSetup({ chainId, safeAddress, newSafe });
+    } catch {
+      return this._notifyUnknownTransaction(chainId, log);
+    }
```

Comment:
Good point. `_notifyNewSafeSetup` (to be renamed) would throw if there is a non-owner management transaction but the decoding (currently) notifies of that further up.

What do you think about removing the `catch` from within `decodeTransactionAdded` and calling it within the `try` instead? I think that would be a bit clearer. cc @hectorgomezv

Reply:
Yes, I think it makes more sense and it will simplify the code 👍🏻

Reply:
I've included the suggested refactor in 110716087c1e6c60867165586081cdbf23c478dc.

Reply:
> Good point. `_notifyNewSafeSetup` (to be renamed) would throw if there is a non-owner management transaction but the decoding (currently) notifies of that further up.

You mean `_mapSafeSetup` right? 

> What do you think about removing the `catch` from within `decodeTransactionAdded` and calling it within the `try` instead? I think that would be a bit clearer. cc @hectorgomezv

There's multiple layers of error handling – the service itself (`AlertsService`) also has a try catch which logs a warning but then returns to the caller 200.

I think in the end it's important to notify the user even on payloads that we do not expect so we should probably not throw as I commented before.

Reply:
> You mean `_mapSafeSetup` right?

Yes. I've reverted my change for now as it has an influence on my [proposed retrieval of the Safe address for unknown transactions](https://github.com/safe-global/safe-client-gateway/pull/923) as well. Refactoring this would make sense there once we are in agreement.